### PR TITLE
move file logger into pic

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h [hunk 1]
```diff
@@ -1375,6 +1375,65 @@ PUBLIC_API SIZE_T getInstrumentedTotalAllocationSize();
...
+ * previously set by the create function.
+ *
+ * @return - STATUS code of the execution
+ */
+PUBLIC_API STATUS freeFileLogger();
+
+/**
+ * Helper macros to be used in pairs at the application start and end
+ */
+#define CREATE_DEFAULT_FILE_LOGGER()                                \
```
Comment (MushMal) [inline src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1429]: I would prefer SET_XXX and RESET_XXX similar to other macros we use.

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h [hunk 2]
```diff
@@ -1375,6 +1375,65 @@ PUBLIC_API SIZE_T getInstrumentedTotalAllocationSize();
...
+/**
+ * File logger error values starting from 0x41300000
+ */
+#define STATUS_FILE_LOGGER_BASE                                             STATUS_UTILS_BASE + 0x01200000
+#define STATUS_FILE_LOGGER_INDEX_FILE_INVALID_SIZE                          STATUS_SEMAPHORE_BASE + 0x00000001
+
+/**
+ * File based logger limit constants
+ */
+#define MAX_FILE_LOGGER_STRING_BUFFER_SIZE          (3 * 1024 * 1024)
```
Comment (MushMal) [inline src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1391]: Perhaps we can bump up the max file logger buffer size to something larger. There is no reason for it to limit the max

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h [hunk 3]
```diff
@@ -1375,6 +1375,65 @@ PUBLIC_API SIZE_T getInstrumentedTotalAllocationSize();
...
+#define MIN_FILE_LOGGER_STRING_BUFFER_SIZE          (10 * 1024)
+#define MAX_FILE_LOGGER_LOG_FILE_COUNT              (10 * 1024)
+
+/**
+ * Default values used in the file logger
+ */
+#define FILE_LOGGER_LOG_FILE_NAME                   "kvsFileLog"
+#define FILE_LOGGER_LAST_INDEX_FILE_NAME            "kvsFileLogIndex"
+#define FILE_LOGGER_STRING_BUFFER_SIZE              (100 * 1024)
```
Comment (MushMal) [inline src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1400]: Bump it to half a meg?

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h [hunk 4]
```diff
@@ -1375,6 +1375,65 @@ PUBLIC_API SIZE_T getInstrumentedTotalAllocationSize();
...
+#define FILE_LOGGER_LOG_FILE_COUNT                  3
```
Comment (MushMal) [inline src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1401]: Same with this? Perhaps make it a little more

File: src/utils/src/FileLogger.c [hunk 5]
```diff
@@ -0,0 +1,250 @@
+/**
+ * Kinesis Video Producer File based logger
+ */
+#define LOG_CLASS "FileLogger"
+#include "Include_i.h"
+
+static PFileLogger gFileLogger = NULL;
```
Comment (MushMal) [inline src/utils/src/FileLogger.c:7]: Important! Make sure remove static keyword here to ensure a single instance for the entire process address space

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h (hunk 1 above)
Comment (chehefen) [reply src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1429]: I dont think SET_XXX is properly convey the purpose of this api. All this api does is create a file logger with default configs.

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h (hunk 2 above)
Comment (chehefen) [reply src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1391]: createFileLogger

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h (hunk 3 above)
Comment (chehefen) [reply src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1400]: if ppl dont want the default setting they can always use createFileLogger.

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h (hunk 4 above)
Comment (chehefen) [reply src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1401]: if ppl dont want the default setting they can always use createFileLogger.

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h (hunk 2 above)
Comment (MushMal) [reply src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1391]: This is a max and in order to change the behavior and make it not fail the customer needs to change this constant.

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h (hunk 3 above)
Comment (MushMal) [reply src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1400]: That sounds good - just need to select a sensible defaults at any stage. Make some calculations and see how much 3 files per 100K lines can hold appx in minutes for debug logs. Select something that can hold a few mins worth of logs

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h (hunk 1 above)
Comment (MushMal) [reply src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1429]: Yeah, it creates and sets. I just wanted to point out the consistency.  Want to make sure that the setup and cleanup are in sync with each other and other declarations we had before. Your call.

File: src/utils/include/com/amazonaws/kinesis/video/utils/Include.h (hunk 2 above)
Comment (chehefen) [reply src/utils/include/com/amazonaws/kinesis/video/utils/Include.h:1391]: this is default max, it's configurable in create file logger

**Pull request merged**